### PR TITLE
feat: round robin queueing

### PR DESCRIPTION
**Pull request by abelanger5**
# Description

Implements a round-robin queueing strategy across concurrency keys. Load is distributed equally between concurrency keys of different groups. 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# What's Changed

- [X] Engine support for `GROUP_ROUND_ROBIN`
- [X] API support for `GROUP_ROUND_ROBIN`
- [X] Go SDK updated
- [X] Python SDK updated
- [X] Typescript SDK updated

### CODE REVIEW

File: internal/repository/prisma/dbsqlc/sqlc.yaml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 version: "2"
 sql:
   - engine: "postgresql"
+    # database:
```
Comment (grutt) [inline internal/repository/prisma/dbsqlc/sqlc.yaml:4]: comment?

File: python-sdk/hatchet_sdk/worker.py [hunk 2]
```diff
@@ -138,6 +138,7 @@ def callback(future : Future):
                 if not errored:
                     # Create an action event
                     try:
+                        print(action)
```
Comment (grutt) [inline python-sdk/hatchet_sdk/worker.py:141]: rm print

File: python-sdk/hatchet_sdk/worker.py [hunk 3]
```diff
@@ -286,8 +287,11 @@ def get_group_key_action_finished_event(self, action : Action, output : str) ->
         except Exception as e:
             logger.error(f"Could not create action finished event: {e}")
             raise e
-
-        event.eventPayload = output
+        
+        try:
+            event.eventPayload = output
+        except Exception as e:
+            event.eventPayload = ""
```
Comment (grutt) [inline python-sdk/hatchet_sdk/worker.py:294]: what is this catch block for? there assignment is throwing?

File: typescript-sdk/examples/concurrency/group-round-robin/concurrency-worker.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      run: async (ctx) => {
+        const { data } = ctx.workflowInput();
+        const { signal } = ctx.controller;
+
+        if (signal.aborted) throw new Error('step1 was aborted');
+
+        console.log('starting step1 and waiting 5 seconds...', data);
+        await sleep(2000);
+
+        if (signal.aborted) throw new Error('step1 was aborted');
```
Comment (grutt) [inline typescript-sdk/examples/concurrency/group-round-robin/concurrency-worker.ts:35]: is this signaling used with round robin?

**Review by grutt — APPROVED** [review]

File: typescript-sdk/examples/concurrency/group-round-robin/concurrency-worker.ts (hunk 4 above)
Comment (abelanger5) [reply typescript-sdk/examples/concurrency/group-round-robin/concurrency-worker.ts:35]: No, this part shouldn't be reached.

File: internal/repository/prisma/dbsqlc/sqlc.yaml (hunk 1 above)
Comment (abelanger5) [reply internal/repository/prisma/dbsqlc/sqlc.yaml:4]: Leaving it for now as it was very helpful in addressing some strange `sqlc` behavior.

**Pull request merged**